01_class_change_page/class-give-page.php

```diff
@@ -0,0 +1,129 @@
+<?php
+ini_set("display_errors", 0);
+error_reporting(E_ALL);
+require_once('submit.php');
+
+// DB接続
+try {
+    $pdo = new PDO($dsn, $user, $password);
+    $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+} catch (Exception $e) {
+    echo "<span class='error'>エラーがありました。</span><br>";
+    echo $e->getMessage();
+    exit();
+}
+
+date_default_timezone_set ('Asia/Tokyo');
+?>
+
+<!DOCTYPE html>
+<html lang="ja">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="change-style.css">
+    <link rel="stylesheet" href="../header.css">
+    <link rel="stylesheet" href="../practices_Bootstrap/css/bootstrap.min.css">
+    <title>授業変更ページ</title>
+</head>
+<body>
+    <?php include('../practices_Bootstrap/index.html'); ?>
+    <main class="main-contents container-fruid ms-2 ms-sm-5 maxWidth">
+        <!-- 行 -->
+        <h1 class="row">授業変更ページ</h1>
+        <!-- 行 -->
+        <h3 class="row">変更内容：譲渡</h3>
+        <div id="container-give" class="block">
+            <!-- 行 -->
+            <p class="before-title row my-3">変更する授業を選択</p>
+            <div class="container-form">
+                <!-- 行 -->
+                <form class="before row align-items-center my-3" action="" method="POST">
+                    <!-- クラス選択 -->
+                    <p class="col-1 m-0">クラス</p>
+                    <select name="class-select" class="selection-area col-2">
+                        <option value="" hidden>クラスを選択</option>
+                        <?php for ($i = 0; $i < count($result_class); $i++){
+                            if ((string)$result_class[$i][0] == $_POST["class-select"]) {
+                                echo "<option value='" . $result_class[$i][0] . "' selected>" . $result_class[$i][1] . "</option>" ;
+                            } else {
+                                echo "<option value='" . $result_class[$i][0] . "'>" . $result_class[$i][1] . "</option>" ;
+                            }
+                        } ?>
+                    </select>
+                    <!-- 日付選択 -->
+                    <p class="col-1 m-0">日付</p>
+                    <input class="col-2" type="date" name="day-select" value="<?php if (!empty($_POST["day-select"])) echo $_POST["day-select"]; ?>">

```

Comment:
[must]
ここですが、XSS脆弱性が出てしまっているのでセキュリティ的にまずいです。(その他 echo などを利用してる所も同様)
HTMLをechoなどで動的にview出力する場合は全て下記のようにエスケープするようにしてください。(特にユーザーからPOSTしたデータを表示する際は絶対にHTMLをエスケープしないとダメです)
このままだと悪意のあるユーザーがこのアプリを利用したときに個人情報抜、クレジットカードの情報ガ抜かれるとか、ウェブサイトやデータの改竄をれたりやられたい放題されます 😭 
世のサービスのでたまにクレカの情報が抜かれたとかニュースになっていますが、だいたいこれが原因だったりします。

今一度、XSSなどのセキュリティについて合わせて学習しましょう！

```suggestion
                    <input class="col-2" type="date" name="day-select" value="<?php if (!empty($_POST["day-select"])) echo htmlspecialchars($_POST["day-select"], ENT_QUOTES, 'UTF-8'); ?>">

```